src/masked-input/types.ts

```diff
@@ -0,0 +1,22 @@
+import { InputGroupProps } from "../input-group";
+
+export interface MaskedInputProps<T, V>
+    extends Omit<InputGroupProps<T, V>, "addon"> {
```

Comment:
the `onBlur` from `InputGroupProps` differs from the one in `InputProps` in that the former doesn't take in an argument and the latter does. if i extend from `InputProps`, the `MaskedInput`'s onBlur callback expects an argument, but the `onBlur` type in `InputGroup` which `MaskedInput` uses doesn't return an argument. so there was a type error which was why i extended from `InputGroupProps`. but anyway i managed to resolve it by setting the event arg to optional, not sure if that's the correct way of handling..

Reply:
I see. It's because onBlur event doesn't always originate from the input element in `InputGroup` 😞

In that case MaskedInput should indeed follow `InputGroupProps`. Sorry for the trouble.

Reply:
on second thought I find leaving the generic here to be weird

can `MaskedInput` specify its own `onBlur` typing where the event is optional?

Reply:
but i still think its more correct to extend from `InputProps` since the MaskedInput specifically uses the **custom** `InputGroup` in which the onBlur event does indeed originates from the input element... 🤔 let me know what you think!

Reply:
agree. it's safe to assert that the blur event is always visible for the custom variant.

perhaps we can just have a comment on why the `?` is needed

Reply:
instead of setting the `event` to an optional argument on `MaskedInput`'s `handleBlur`, i fixed the `onBlur` prop type on `InputGroup` to take in an optional `event` argument and added a comment on why that is needed